### PR TITLE
implemented custom data validation and customglobalexceptionhandler

### PR DESCRIPTION
**Pull request by nastasiiaKulinich**

### CODE REVIEW

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -37,14 +38,14 @@ public BookDto getBookById(@PathVariable Long id) {
...
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
 
     @ResponseStatus(HttpStatus.OK)
     @PutMapping("/{id}")
     public BookDto updateBookById(@PathVariable Long id,
-                              @RequestBody CreateBookRequestDto requestDto) {
+                              @RequestBody @Valid CreateBookRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/controller/BookController.java:51]: Also, it's better to add `@PathVariable @Positive Long id` to getBookById method.
Don't forget to add `@Validated` under the controller

File: src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
 
+import com.example.bookstore.validation.Isbn;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java]: For String object it's better to use `@NotBlank`

File: src/main/java/com/example/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    private String getErrorMessage(ObjectError e) {
+        if (e instanceof FieldError fieldError) {
+            String field = fieldError.getField();
+            String message = e.getDefaultMessage();
+            return field + " " + message;
+        }
+        return e.getDefaultMessage();
+    }
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/exception/CustomGlobalExceptionHandler.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's make it better

**Review by okuzan — APPROVED** [review]

**Pull request merged**